packages/checkbox/theme/lumo/vaadin-checkbox-styles.js

```diff
@@ -26,6 +26,14 @@ registerStyles(
       --_focus-ring-color: var(--vaadin-focus-ring-color, var(--lumo-primary-color-50pct));
       --_focus-ring-width: var(--vaadin-focus-ring-width, 2px);
       --_selection-color: var(--vaadin-selection-color, var(--lumo-primary-color));
+      --_helper-spacing: var(--vaadin-input-field-helper-spacing, 0.4em);
+      --_invalid-background: var(--vaadin-input-field-invalid-background, var(--lumo-error-color-10pct));
```

Comment:
Regarding the missing hover style for the invalid state, it is kind of noticable. I guess a simple hack would be to calculate the exact color that results from adding the hover color to the background color, and use that as background instead. But that also makes theming more difficult. There might also be some newer CSS features that allow combining two colors (there's color-mix at least). Otherwise not a blocker if we can't find a reasonable solution.

Reply:
Seems like a decent solution, then it's still customizable with the theme variables. Support for it in Safari is fairly new (16.2), but I think for a non-critical feature like this not supporting older versions is fine.

Reply:
Hmm, yeah, we're in kind of a tricky situation with Safari support, with V24.x being stuck on 15.0 from back in September _2021_. Normally I'd say we have to stick to that until V25 comes along, but as you say, this is a very minor detail so the effect on users with older browser is negligible. But we need to make sure it degrades gracefully instead of making the box transparent or black or something.

Reply:
Also:

- Maybe the invalid box color should be a bit darker in general, similarly to how the normal box color is 10% darker/opaquer than the text-field background color? It does look at bit too faint against a white background (possibly on the verge of invisible on bad monitors).
- The optional border should use the error color in invalid state (check how text-field does that).

Reply:
> The optional border should use the error color in invalid state (check how text-field does that).

Done.

Reply:
> Maybe the invalid box color should be a bit darker in general

Currently, we only have Lumo error colors ` --lumo-error-color-10pct` and ` --lumo-error-color-50pct` so no error color with `20pct` which would correspond to `--lumo-contrast-20pct`. I wonder if we should add that

Reply:
Update: added `background-image` with gradient to make hover state darker:

```css
    :host([invalid]:hover) [part='checkbox'] {
      background-image: linear-gradient(var(--_invalid-background) 0%, var(--_invalid-background) 100%);
    }
```